### ISSUE TITLE
New rule or option to remove unnecessary indentation in multiline paragraphs

### ISSUE THREAD

**Issue by palto42** [rvben/rumdl#747]
### Problem or motivation

With MD013 reflow mode "normal", leading space indentation is kept in case of reformatting, while it is removed in all other reflow modes.

Example text, which is rendered as single paragraph:
```
This is some paragraph
 with line breaks
  and indentation.
```

This should be re-formatted as
```
This is some paragraph
with line breaks
and in indentation.
```

### Proposed solution

Possible solutions:

1. Modify the reflow mode normal to remove unnecessary leading spaces in multiline paragraphs
2. Add an option to MD013 to enable removal of unnecessary leading spaces
3. Add new rule to check for unnecessary leading space and remove them in case of reformatting

### Feature category

New lint rule

### Example markdown

```markdown

```

### Contribution

- [ ] I would be willing to submit a PR to implement this feature

**Comment by rvben** [comment]
Thanks for the report @palto42, and for laying out three options - that framing is what settled the design.

Shipped in v0.2.44 as **MD085** (`paragraph-continuation-indent`), your third option: a separate rule, so you get it without opting into reflow.

```toml
[global]
extend-enable = ["MD085"]
```

`rumdl fmt` turns your example into exactly the output you asked for.

Two deliberate limits: the first line of a paragraph is never touched (its indentation decides which block the paragraph is), and a continuation line is left alone when removing the indent would make it open a new block (`#`, `>`, `-`, a digit, a fence). Four-space continuation lines *are* rewritten, since an indented code block cannot interrupt a paragraph.

Docs: <https://rumdl.dev/md085/>
